### PR TITLE
Implement policy verification in DFA

### PR DESCRIPTION
**Pull request by bgogul**
This PR provides the missing connection between policy and DFA. 

- Generate DFA checks and claims by invoking the policy translator. (This step will eventually move to compile time.)
- Incorporate the checks into the recipe, run DFA, and validate the checks.

If the recipe violates any aspect of the policy, a `PolicyViolation` exception is thrown. It is enough to review the second commit eb349ddaade5f07a63ba26d18a615598c3a80bf4. This PR is dependent on https://github.com/PolymerLabs/arcs/pull/5785.

### CODE REVIEW

File: java/arcs/core/analysis/PolicyVerifier.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import arcs.core.policy.PolicyOptions
+import arcs.core.policy.PolicyViolation
+import arcs.core.policy.translatePolicy
+
+/** A class to verify that recipes are compliant with a policy. */
+class PolicyVerifier(val options: PolicyOptions) {
+    /**
+     * Verifies that the recipe is compliant with the given policy.
+     *
+     * @throws [arcs.core.policy.PolicyViolation] if policy is violated by the recipe.
```
Comment (csilvestrini) [inline java/arcs/core/analysis/PolicyVerifier.kt:31]: No need to mention the package here

File: java/arcs/core/analysis/PolicyVerifier.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     */
+    public fun verifyPolicy(recipe: Recipe, policy: Policy): Boolean {
+        // TODO(bgogul): This should be moved to the compilation step.
```
Comment (csilvestrini) [inline java/arcs/core/analysis/PolicyVerifier.kt:34]: Bug number?

File: java/arcs/core/analysis/PolicyVerifier.kt [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import arcs.core.policy.Policy
+import arcs.core.policy.PolicyConstraints
+import arcs.core.policy.PolicyOptions
+import arcs.core.policy.PolicyViolation
+import arcs.core.policy.translatePolicy
+
+/** A class to verify that recipes are compliant with a policy. */
+class PolicyVerifier(val options: PolicyOptions) {
+    /**
+     * Verifies that the recipe is compliant with the given policy.
```
Comment (csilvestrini) [inline java/arcs/core/analysis/PolicyVerifier.kt:29]: Mention that it returns true?

File: java/arcs/core/analysis/PolicyVerifier.kt [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+                    it.predicate
+                )
+            }
+            ?.let { claims -> InformationFlow.IngressInfo(handleNode, claims) }
+    }
+
+    /**
+     * Returns the egress checks.
+     *
+     * @throws [arcs.core.policy.PolicyViolation] if recipe violates egress requirements.
```
Comment (csilvestrini) [inline java/arcs/core/analysis/PolicyVerifier.kt:95]: No package needed

File: java/arcs/core/analysis/PolicyVerifier.kt (hunk 3 above)
Comment (bgogul) [reply java/arcs/core/analysis/PolicyVerifier.kt]: done.

File: java/arcs/core/analysis/PolicyVerifier.kt (hunk 2 above)
Comment (bgogul) [reply java/arcs/core/analysis/PolicyVerifier.kt]: done.

**Pull request merged**